core/src/main/java/org/elasticsearch/cluster/routing/allocation/MoveDecision.java

```diff
@@ -129,44 +109,33 @@ public boolean cannotRemain() {
     }
 
     /**
-     * Gets the individual node-level decisions that went into making the final decision as represented by
-     * {@link #getFinalDecisionType()}.  The map that is returned has the node id as the key and a {@link NodeAllocationResult}.
-     */
-    @Nullable
-    public Map<String, NodeAllocationResult> getNodeDecisions() {
-        return nodeDecisions;
-    }
-
-    /**
      * Gets the final explanation for the decision to move a shard.
      */
     @Override
     public String getExplanation() {
         String explanation;
         if (cannotRemain() == false) {
             explanation = "can remain on its current node";
-        } else if (getFinalDecisionType() == Type.YES) {
+        } else if (getDecisionType() == Type.YES) {
             explanation = "shard cannot remain on this node and is force-moved to another node";
-        } else if (getFinalDecisionType() == Type.THROTTLE) {
+        } else if (getDecisionType() == Type.THROTTLE) {
             explanation = "shard cannot remain on this node but is throttled on moving to another node";
         } else {
-            assert getFinalDecisionType() == Type.NO;
+            assert getDecisionType() == Type.NO;
             explanation = "cannot move shard to another node, even though it is not allowed to remain on its current node";
         }
         return explanation;
     }
 
     @Override
-    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+    public void innerToXContent(XContentBuilder builder, Params params) throws IOException {
         super.toXContent(builder, params);
         builder.startObject("can_remain_decision");
         {
             builder.field("decision", canRemainDecision.type().toString());
             canRemainDecision.toXContent(builder, params);
         }
         builder.endObject();
-        nodeDecisionsToXContent(builder, params, nodeDecisions);
```

Comment:
this reorders the output. I think that the StreamOutput/StreamInput can call super() but that the toXContent should be completely done by the subclass so that we can render output in the optimal way. I would for example prefer to see the can_remain_decision **before** the node decisions where to allocate.